### DISCUSSION TITLE
Jadual Tugasan Imam/Bilal

### DISCUSSION THREAD

**Discussion by rohaizanr** [Dev4w4n/e-masjid.my#136] [Ideas]
Jadual imam bertugas pada waktu tu. Imam dan bilal yang bertugas.

Buat auto generate imam bertugas, dan sekaligus boleh edit.

**Comment by rohaizanr** [comment]
kita perlukan UI mockup dulu. ideas?

**Comment by AimanFariz** [comment]
Saya ada cadangan untuk UI page ni. Saya sertakan sekali wireframe yg saya ada buat di Figma:

![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/08b806fe-f826-478f-ab24-808f4ae6488c)
Yang ni default page untuk jadual imam and bilal,

![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/be651387-125f-4924-a473-2637d916fe58)
Pengguna boleh select nak tgk jadual by hari, minggu, or bulan.

Kalau by minggu, then jadual akan tunjuk hari. Kalau pengguna select Hari, then jadual akan tunjuk 5 waktu solat fardhu and siapa yang bertugas.

![image](https://github.com/Dev4w4n/e-masjid.my/assets/99475959/a1171468-9584-402e-b8b8-cc69058c4389)
Ni cadangan saya kalau pengguna/petugas masjid nak tambah shif imam atau bilal. Butang '+' tu akan appear waktu hover.

Butang dropdown 'Kumpulan' dekat belah kanan atas tu, pengguna boleh pilih nak tengok jadual imam sahaja atau bilal sahaja atau semua. By default, pengguna akan tengok jadual semua, tapi, akan dibahagikan ikut kategori imam dan bilal.

Satu lagi, ini wireframe untuk page 'Jadual Keseluruhan', jadi boleh tengok imam2 dan bilal2 lain punya jadual juga. Kalau pengguna pilih 'Jadual Saya', dia akan tengok jadual bertugas dia seorang saja. Untuk yang ni, saya belum wireframe lagi. Saya nak minta pandangan yang lain dulu sebelum saya teruskan.

Sekian.

**Comment by rohaizanr** [comment]
Idea yang baik. terima kasih @AimanFariz 
 
Saya akan cuba dapatkan contoh Laporan Jadual Imam/Bilal dari masjid saya.

Biasanya, di masjid saya Pengerusi yang akan atur jadual ini, dan dia akan print jadual tersebut dan lekatkan di Papan Kenyataan masjid.
